runtime/lua/vim/treesitter/dev.lua

```diff
@@ -100,19 +100,39 @@ function TSTreeView:new(bufnr, lang)
   -- the primary tree that contains that root. Add a mapping from the node in the primary tree to
   -- the root in the child tree to the {injections} table.
   local root = parser:parse(true)[1]:root()
-  local injections = {} ---@type table<integer,table>
-  for _, child in pairs(parser:children()) do
-    child:for_each_tree(function(tree, ltree)
-      local r = tree:root()
-      local node = root:named_descendant_for_range(r:range())
+  local injections = {} ---@type table<integer,{lang: string, root: TSNode}>
+
+  ---@param node TSNode
+  ---@return number
+  local function node_length(node)
+    local _, _, start_byte = node:start()
+    local _, _, end_byte = node:end_()
+    return end_byte - start_byte
+  end
+  parser:for_each_tree(function(parent_tree, parent_ltree)
+    local parent = parent_tree:root()
+    for _, child in pairs(parent_ltree:children()) do
+      local r = child:trees()[1]:root()
```

Comment:
This seems unsafe but I don't know. I'm just expecting the first tree to be what I expected

Reply:
It breaks, so I attempted it a different way